src/components/date-picker/date-picker.tsx

```diff
@@ -195,26 +125,63 @@ export class DatePicker {
                     helperText={this.helperText}
                     required={this.required}
                     value={this.formattedValue}
-                    onChange={this.handleChange}
+                    onFocus={this.showPortalElement}
+                    ref={(el) => (this.textField = el)}
                 />
+                <limel-portal
+                    containerId={this.portalId}
+                    visible={this.showPortal}
+                >
+                    <limel-flatpickr-adapter
+                        type={this.type}
+                        value={this.value}
+                        ref={(el) => (this.datePickerCalendar = el)}
+                        onChange={this.handleCalendarChange}
+                    />
+                </limel-portal>
             </div>
         );
     }
 
-    @Listen('resize', { target: 'window' })
-    public resizeEvent() {
-        this.picker.init(this.input, this.container, this.value);
-    }
-
     @Watch('value')
     protected onValueChange(newValue, oldValue) {
-        if (newValue !== oldValue) {
-            this.formattedValue = this.picker.formatDate(newValue);
+        if (newValue !== oldValue && newValue !== this.formattedValue) {
+            this.formattedValue = this.dateFormatter.formatDateByType(
+                newValue,
+                this.type
+            );
         }
     }
 
-    private handleChange(event) {
+    private showPortalElement(event) {
+        this.showPortal = true;
+        const inputElement = this.textField.shadowRoot.querySelector('input');
+        setTimeout(() => {
+            this.datePickerCalendar.inputElement = inputElement;
+        });
+        event.stopPropagation();
+
+        document.addEventListener('mousedown', this.eventListener, { passive: true });
+    }
+
+    private hidePortalElement() {
+        setTimeout(() => {
+            this.showPortal = false;
+        })
+        document.removeEventListener('mousedown', this.eventListener);
+    }
+
+    private handleCalendarChange(event) {
+        const date = event.detail;
+        this.formattedValue = this.dateFormatter.formatDateByType(
+            date,
+            this.type
+        );
         event.stopPropagation();
-        this.formattedValue = event.detail;
+        if (this.type !== 'datetime' && this.type !== 'time') {
+            this.textField.blur();
```

Comment:
I think this means the next tab press would focus the first focusable element on the page. Not an issue at the moment, since these aren't keyboard accessible anyway, but just mentioning it for the future.

Reply:
Bump

Reply:
I think I need some help with solving this :)